packages/validator/src/util/externalSignerClient.ts

```diff
@@ -157,20 +164,48 @@ export async function externalSignerPostSignature(
 export async function externalSignerUpCheck(remoteUrl: string): Promise<boolean> {
   const res = await fetch(`${remoteUrl}/upcheck`, {
     method: "GET",
-    headers: {"Content-Type": "application/json"},
+    headers: {Accept: MediaType.json},
   });
 
-  const data = await handlerExternalSignerResponse<{status: string}>(res);
+  const data = await handleExternalSignerResponse<{status: string}>(res);
   return data.status === "OK";
 }
 
-async function handlerExternalSignerResponse<T>(res: Response): Promise<T> {
+async function handleExternalSignerResponse<T>(res: Response): Promise<T> {
   if (!res.ok) {
     const errBody = await res.text();
-    throw Error(`${errBody}`);
+    throw Error(errBody ? getErrorMessage(errBody) : res.statusText);
+  }
+
+  const contentType = res.headers.get("content-type");
+  if (contentType === null) {
+    throw Error("No Content-Type header found in response");
+  }
+
+  const mediaType = contentType.split(";", 1)[0].trim().toLowerCase();
+
+  if (mediaType !== MediaType.json) {
+    throw Error(`Unsupported response media type: ${mediaType}`);
+  }
+
+  try {
+    return (await res.json()) as T;
+  } catch (e) {
+    throw Error(`Invalid json response: ${(e as Error).message}`);
   }
+}
 
-  return JSON.parse(await res.text()) as T;
+function getErrorMessage(errBody: string): string {
+  try {
+    const errJson = JSON.parse(errBody) as {message: string};
```

Comment:
I don't think it's a good idea to treat error response as json if it's not in the spec. Even if the error response is actually json, you are not guaranteed that there is a property named `message` (it could be `msg`, `error` or even `errorMsg` ...) and the result will be worse than if you had included the response content as plain text. You may also need to check the `content-type' of the error response before treating it as json. I know there is a try/catch block, but why do this if it's not in the spec?

Reply:
Ok fair, we would have to check content type and response format is unclear. Just assumed sane format similar to beacon api. Although for errors is not as critical to check proper format and have fallback logic via try/catch.

Let's keep error handling as is, the Diva team is working on the spec will ask them about it and see if we can standardize this a bit better.